cli/util/file_test.go

```diff
@@ -53,3 +54,50 @@ func TestTFDirectories(t *testing.T) {
 		})
 	}
 }
+
+func TestFindFilesWithPattern(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		pattern string
+		want    []string
+		wantErr bool
+	}{
+		{
+			name:    "pattern for metadata files",
+			path:    "",
+			pattern: `.*/metadata(?:.display)?.yaml`,
+			want: []string{
+				"../testdata/bpmetadata/content/examples/acm/acm-terraform-blog-part1/terraform/metadata.yaml",
+				"../testdata/bpmetadata/content/examples/acm/metadata.display.yaml",
+				"../testdata/bpmetadata/content/examples/acm/metadata.yaml",
```

Comment:
Might want to add some test cases for "strange" situations:

"../testdata/bpmetadata/content/examples/acm/metadata.yaml.invalid",
"../testdata/bpmetadata/content/examples/acm/metadata.yaml/file",

Reply:
Added.